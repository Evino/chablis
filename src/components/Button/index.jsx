import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { colors, sizes, fontSizes, fontWeights } from '../../styles/variables';

const ActionButton = styled.button`
  max-width: 300px;
  width: 100%;
  min-height: 50px;
  background: ${colors.PRIMARY};
  font-size: ${fontSizes.SMALL};
  font-weight: ${fontWeights.MEDIUM};
  color: ${colors.WHITE};
  border: 1px solid ${colors.PRIMARY};
  outline: 0;
  border-radius: ${sizes.RADIUS};
  cursor: pointer;
  display: block;
  transition: .3s;

  &:hover {
    background: ${colors.PRIMARY_DARKEN};
    border-color: ${colors.PRIMARY_DARKEN};
    color: ${colors.WHITE};
  }

  &:disabled {
    cursor: default;
    color: ${colors.DARK_GREY};
    background: ${colors.LIGHTEST_GREY};
    border-color: ${colors.LIGHTEST_GREY};
  }

  ${props => props.thin && css`
    height: 30px;
  `}

  ${props => props.fluid && css`
    max-width: ${sizes.FULL};
  `}

  ${props => props.secondary && css`
    background: transparent;
    color: ${colors.PRIMARY};
  `}

  ${props => props.secondary && props.disabled && css`
    background: ${colors.WHITE} !important;
    border-color: ${colors.DARK_GREY} !important;
  `}
`;

export default class Button extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    fluid: PropTypes.bool,
    secondary: PropTypes.bool,
    disabled: PropTypes.bool,
    thin: PropTypes.bool,
  }

  static defaultProps = {
    fluid: false,
    secondary: false,
    disabled: false,
    thin: false,
  }

  render() {
    return (
      <ActionButton {...this.props}>{this.props.children}</ActionButton>
    );
  }
}
