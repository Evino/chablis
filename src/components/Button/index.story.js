/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './';

const action = param => {
  alert(param);
}

storiesOf('Button', module)
  .addWithInfo('Primary Button', () => (
    <Button
      onClick={action.bind(this, 'clicked')}
    >
      Primary Button
    </Button>
  ))
  .addWithInfo('Fluid Button', () => (
    <Button
      onClick={action.bind(this, 'clicked')}
      fluid
    >
      Fluid Button
    </Button>
  ))
  .addWithInfo('Thin Button', () => (
    <Button
      onClick={action.bind(this, 'clicked')}
      thin
    >
      Thin Button
    </Button>
  ))
  .addWithInfo('Disabled Button', () => (
    <Button
      onClick={action.bind(this, 'clicked')}
      disabled
    >
      Disabled Button
    </Button>
  ))
  .addWithInfo('Secondary Button', () => (
    <Button
      onClick={action.bind(this, 'clicked')}
      secondary
    >
      Secondary Button
    </Button>
  ));