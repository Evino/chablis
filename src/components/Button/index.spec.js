import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './';

const onClick = jest.fn();
const element = <Button onClick={onClick}>Button</Button>;
const component = mount(element);

const click = () => {
  component.find('button').simulate('click');
};

describe('Button', () => {
  it('should render a button', () => {
    const tree = renderer
      .create(element)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should allow client`s click', () => {
    click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should not allow client`s click', () => {
    component.setProps({ disabled: true });
    click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
