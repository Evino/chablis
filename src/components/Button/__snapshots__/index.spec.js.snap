// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
<button
  className="Button__ActionButton-sc-1efou2l-0 blcmaT"
  disabled={false}
  onClick={[MockFunction]}
>
  Button
</button>
`;
